components/arm/universalrobots/ur.go

```diff
@@ -214,19 +217,28 @@ func URArmConnect(ctx context.Context, conf resource.Config, logger golog.Logger
 						return
 					}
 					logger.Debug("attempting to reconnect to ur arm dashboard")
+					time.Sleep(1 * time.Second)
 					connDashboard, err = d.DialContext(cancelCtx, "tcp", newArm.host+":29999")
 					if err == nil {
 						newArm.mu.Lock()
 						newArm.dashboardConnection = connDashboard
+						newArm.isConnected = true
 						newArm.mu.Unlock()
 						break
+					} else {
+						newArm.mu.Lock()
+						newArm.isConnected = false
+						newArm.mu.Unlock()
 					}
 					if !goutils.SelectContextOrWait(cancelCtx, 1*time.Second) {
 						return
 					}
 				}
 			} else if err != nil {
 				logger.Errorw("dashboard reader failed", "error", err)
+				newArm.mu.Lock()
+				newArm.isConnected = false
+				newArm.mu.Unlock()
```

Comment:
@npmenard I added a check in the test like you suggested to check that `isConnected` was false after disconnection, and it failed on one random run and the errors printed were an EOF error and this "dashboard reader failed" error, so I just added these lines to set `isConnected` to false. Just want to double check with you that that seems valid?

Reply:
yes